### PR TITLE
Fix sbt startup when running sbt-allsources.sh

### LINKED ISSUE #5193: sbt-allsources.sh build error
## steps
Follow the steps in https://github.com/sbt/sbt/blob/develop/DEVELOPING.md#instruction-to-build-all-modules-from-source

## problem

running `./sbt-allsources.sh` results in:

```
sbt-modules/zinc/build.sbt:425: error: missing argument list for method addSbtUtilLogging in object Dependencies
Unapplied methods are only converted to functions when a function type is expected.
You can make this conversion explicit by writing `addSbtUtilLogging _` or `addSbtUtilLogging(_)` instead of `addSbtUtilLogging`.
  .configure(addSbtIO, addSbtUtilLogging, addSbtUtilRelation)
                       ^
sbt-modules/zinc/build.sbt:425: error: missing argument list for method addSbtUtilRelation in object Dependencies
Unapplied methods are only converted to functions when a function type is expected.
You can make this conversion explicit by writing `addSbtUtilRelation _` or `addSbtUtilRelation(_)` instead of `addSbtUtilRelation`.
  .configure(addSbtIO, addSbtUtilLogging, addSbtUtilRelation)
                                          ^
sbt-modules/zinc/build.sbt:316: error: not found: type IncompatibleSignatureProblem
        exclude[IncompatibleSignatureProblem]("sbt.internal.inc.schema.Problem.unapply"),
                ^
sbt-modules/zinc/build.sbt:317: error: not found: type IncompatibleSignatureProblem
        exclude[IncompatibleSignatureProblem]("sbt.internal.inc.schema.Position.unapply"),
                ^
sbt-modules/zinc/build.sbt:318: error: not found: type IncompatibleSignatureProblem
        exclude[IncompatibleSignatureProblem]("sbt.internal.inc.schema.AnalyzedClass.unapply"),
                ^
sbt-modules/zinc/build.sbt:319: error: not found: type IncompatibleSignatureProblem
        exclude[IncompatibleSignatureProblem]("sbt.internal.inc.schema.AnalyzedClass.unapply"),
                ^
sbt-modules/zinc/build.sbt:320: error: not found: type IncompatibleSignatureProblem
        exclude[IncompatibleSignatureProblem]("sbt.internal.inc.schema.Version.values"),
                ^
sbt-modules/zinc/build.sbt:321: error: not found: type IncompatibleSignatureProblem
        exclude[IncompatibleSignatureProblem]("sbt.internal.inc.schema.Position.unapply"),
```

## expectation

Show sbt prompt

### PR DESCRIPTION
**Pull request by regadas**
This an attempt to fix #5193; I'm able to run `./sbt-allsources.sh` now.

### CODE REVIEW

**Review by eed3si9n — APPROVED** [review]
Thanks @regadas!

File: project/plugins.sbt [hunk 1]
```diff
@@ -8,3 +8,4 @@ addSbtPlugin("de.heikoseeberger" % "sbt-header" % "3.0.2")
 addSbtPlugin("com.eed3si9n" % "sbt-buildinfo" % "0.9.0")
 addSbtPlugin("com.lightbend" % "sbt-whitesource" % "0.1.14")
 addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.9")
+addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.6.1")
```
Comment (eed3si9n) [inline project/plugins.sbt:11]: Looks like this mima is reporting a bunch of signature problem.
I'm guessing most are false positives or internal changes that we can ignore. Could you add exclude rules?

**Review by eed3si9n — CHANGES_REQUESTED** [review]

File: project/plugins.sbt (hunk 1 above)
Comment (regadas) [reply project/plugins.sbt:11]: Hi @eed3si9n sure!

Comment (regadas) [reply project/plugins.sbt:11]: Hi @eed3si9n let me know wdyt!

**Review by eed3si9n — APPROVED** [review]
LGTM pending Travis CI

**Pull request merged**